frb_codegen/src/utils.rs

```diff
@@ -12,15 +12,69 @@ pub fn mod_from_rust_path(code_path: &str, crate_path: &str) -> String {
         .replace('/', "::")
 }
 
-pub fn with_changed_file<F: FnOnce() -> anyhow::Result<()>>(
+pub fn with_changed_file(
     path: &str,
     append_content: &str,
-    f: F,
+    generate_c_from_rust_func: impl FnOnce() -> anyhow::Result<()>,
```

Comment:
Why change `with_changed_file` and let it be aware of what is happening inside the callback? 

"with_changed_file" was designed to be a very general utility function, which changes some file content (while do not know what is that content) and run callback and change back content.

Reply:
I have to say I didn't want to change it before 7ac242970f7992280b739532dc759c91c596bae2, but the CI lint just told me there are too many arguments after I introduce `exclude_symbols` (which I have to), I didn't find a better way, except for separating the `f:F`.

Reply:
For https://github.com/fzyzcjy/flutter_rust_bridge/commit/7ac242970f7992280b739532dc759c91c596bae2#diff-92a01e35fcd86edac10d9dfa11ed17ec30ed1fdf565b37934105fb66e4f4f0a5L28, just create a struct as argument.

```
struct BindgenRustToDartArg{

    rust_crate_dir: &str,
    c_output_path: &str,
    dart_output_path: &str,
    dart_class_name: &str,
    c_struct_names: Vec<String>,
    exclude_symbols: Vec<String>,
    llvm_install_path: &[String],
    llvm_compiler_opts: &str,
}

pub fn bindgen_rust_to_dart(arg: BindgenRustToDartArg) -> anyhow::Result<()> {

Reply:
> For [7ac2429#diff-92a01e35fcd86edac10d9dfa11ed17ec30ed1fdf565b37934105fb66e4f4f0a5L28](https://github.com/fzyzcjy/flutter_rust_bridge/commit/7ac242970f7992280b739532dc759c91c596bae2#diff-92a01e35fcd86edac10d9dfa11ed17ec30ed1fdf565b37934105fb66e4f4f0a5L28), just create a struct as argument.
> 
> ```
> struct BindgenRustToDartArg{
> 
>     rust_crate_dir: &str,
>     c_output_path: &str,
>     dart_output_path: &str,
>     dart_class_name: &str,
>     c_struct_names: Vec<String>,
>     exclude_symbols: Vec<String>,
>     llvm_install_path: &[String],
>     llvm_compiler_opts: &str,
> }
> 
> pub fn bindgen_rust_to_dart(arg: BindgenRustToDartArg) -> anyhow::Result<()> {
> ```

Fixed in d4575dc056dad1167a62ed7e81f06703130370df.